crates/core/tedge/src/cli/connect/c8y_direct_connection.rs

```diff
@@ -0,0 +1,94 @@
+use super::{BridgeConfig, ConnectError};
+use rumqttc::{
+    self, certs, pkcs8_private_keys, Client, Event, Incoming, MqttOptions, QoS, Transport,
+};
+use rustls_0_19::ClientConfig;
+use std::{fs::File, io::BufReader, thread, time::Duration};
+
+// Connect directly to the c8y cloud over mqtt and publish device create message.
+pub fn create_device_with_direct_connection(
+    bridge_config: &BridgeConfig,
+) -> Result<(), ConnectError> {
+    const DEVICE_ALREADY_EXISTS: &[u8] = b"41,100,Device already existing";
+    let address = bridge_config.address.clone();
+    let host: Vec<&str> = address.split(":").collect();
+
+    let mut mqtt_options = MqttOptions::new(bridge_config.remote_clientid.clone(), host[0], 8883);
+    mqtt_options.set_keep_alive(std::time::Duration::from_secs(10));
+
+    let mut client_config = ClientConfig::new();
+    // Use rustls-native-certs to load root certificates from the operating system.
+    client_config.root_store =
+        rustls_native_certs::load_native_certs().expect("Failed to load platform certificates.");
+
+    // Load the device key and certificates
+    let f = File::open(bridge_config.bridge_keyfile.clone())?;
+    let mut key_reader = BufReader::new(f);
+    let result = pkcs8_private_keys(&mut key_reader);
+    let key_chain: Vec<rustls_0_19::PrivateKey> = match result {
+        Ok(key) => key,
+        Err(_) => {
+            return Err(ConnectError::RumqttcPrivateKey);
+        }
+    };
+    let pvt_key = key_chain.first().unwrap().clone();
+
+    let f = File::open(bridge_config.bridge_certfile.clone())?;
+    let mut cert_reader = BufReader::new(f);
+    let result = certs(&mut cert_reader);
+    let cert_chain: Vec<rustls_0_19::Certificate> = match result {
+        Ok(cert) => cert,
+        Err(_) => {
+            return Err(ConnectError::RumqttcCertificate);
+        }
+    };
+    let _ = client_config.set_single_client_cert(cert_chain, pvt_key);
+
+    mqtt_options.set_transport(Transport::tls_with_config(client_config.into()));
+
+    let (mut client, mut connection) = Client::new(mqtt_options, 10);
+    let device_id = bridge_config.remote_clientid.clone();
+    thread::spawn(move || publish_device_create_reqests(&mut client, &device_id));
```

Comment:
We got too many bugs with this kind of event loop independent of the control thread. 
Instead of spawning a thread to publish messages, do something along the lines of what is done in `check_device_status_c8y`. I.e. an event loop that wait for a suback for `s/e` the before sending requests to `s/us`. And sending a second request only after the first one got no responses.

Reply:
done